tests/unit/s2n_self_talk_session_resumption_test.c

```diff
@@ -554,6 +554,93 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_io_pair_close(&io_pair));
     };
 
+    /* Test: s2n_connection_get_session should not return a ticket before it has
+     * been received if TLS 1.3 was negotiated.
+     *
+     * This behavior differs between TLS 1.2 and TLS 1.3, so negotiate TLS 1.3 with
+     * a TLS 1.2 session ticket and assert that the TLS 1.3 connection still
+     * behaves correctly.
```

Comment:
I wonder if instead of duplicating the previous test + adding on an extra get/set session, you should just modify the previous test. They should be almost identical except that final step.